### DISCUSSION TITLE
Does Laika support / generate sitemap and rss feeds?

### DISCUSSION THREAD

**Discussion by softinio** [planet42/Laika#356] [Q&A] [answered]
Does Laika support / generate sitemap and rss feeds? For following a site using an rss reader. 

Thanks.

**Comment by jenshalm** [comment, marked as answer]
No, it does not support this out of the box, mostly because it has not been requested even once in the 10 years Laika exists. However, the rich API of the library should make it fairly straightforward to add this functionality yourself when needed. You can process the AST between parsing and rendering to inspect documents, their titles and dates, etc.

**Reply by softinio** [reply]
Thanks! [will do what you suggested](https://github.com/softinio/scalanews/issues/4).

Laika is awesome thank you for creating it.

FYI this is what I built with Laika that I want rss support for: https://www.scalanews.net/ :-) Unusual use case :-)

**Reply by alexandru** [reply]
👋 The use-case isn't unusual, any website that has the notion of a stream of updates (e.g., release notes or announcements) needs one or multiple RSS feeds.

And it isn't just for old timers using feed readers, such as myself, but also for automating the retrieval of such news from other services. E.g., you could automate, via IFTTT, the publishing of a Twitter/LinkedIn/Mastodon post whenever a news article gets published. And ScalaNews itself, exemplified here, uses RSS feeds to automatically extract articles related to Scala from various websites.

Websites not able to do RSS basically don't exist on the web, in my opinion, as they rely on closed social networks and annoying email newsletters for broadcasting, so I hope it will be considered in the future.

--

PS: Laika does look good, thanks for your work ❤️
